[SDK] Support multi signer when using `multiSigPublicKey.getSigner()`, now only support a single signer.

rzexin opened the pull request:
[SDK] Support multi signer when using `multiSigPublicKey.getSigner()`, now only support a single signer.

## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK: 
- [x] Typescript SDK:

hayes-mysten commented:
This limitation was intentional by request of our cryptography team.  I think the assumption was that multisig would be used in one of 2 ways:

1. For 1 of N type signatures, where a single key is sufficient for signing
2. For accounts where signatures from multiple keys are required, but those keys would not exist in the same application

For the second case, the expectation was that `combinePartialSignatures` would be used instead with signatures collected from the various signers individually.

This restriction does feel a bit artificial to me, and I am curious if there are use cases we aren't considering, but my understanding is that the intention was to make it slightly less convenient for the flows that use multisig in an unexpected way

cc @joyqvq

rzexin commented:
> This limitation was intentional by request of our cryptography team. I think the assumption was that multisig would be used in one of 2 ways:
> 
> 1. For 1 of N type signatures, where a single key is sufficient for signing
> 2. For accounts where signatures from multiple keys are required, but those keys would not exist in the same application
> 
> For the second case, the expectation was that `combinePartialSignatures` would be used instead with signatures collected from the various signers individually.
> 
> This restriction does feel a bit artificial to me, and I am curious if there are use cases we aren't considering, but my understanding is that the intention was to make it slightly less convenient for the flows that use multisig in an unexpected way
> 
> cc @joyqvq

@hayes-mysten Yeah, got it~ and therefore, I have modified the function signature from `getSigner(...signers: [signer: Signer])` to `getSigner(signer: Signer)` to avoid ambiguity.

and then, I have added function `getSigners` for `MultiSigPublicKey` to support multi signer to provide more choices for developers. Please take a look to see if the changes are necessary. Thanks!
